ner_v1/detectors/base_detector.py

```diff
@@ -0,0 +1,193 @@
+import abc
+from ner_v1.language_utilities.constant import ENGLISH_LANG
+from ner_v1.constant import (FROM_STRUCTURE_VALUE_VERIFIED, FROM_STRUCTURE_VALUE_NOT_VERIFIED, FROM_MESSAGE,
+                             FROM_FALLBACK_VALUE, ORIGINAL_TEXT, ENTITY_VALUE, DETECTION_METHOD, ENTITY_VALUE_DICT_KEY)
+from ner_v1.language_utilities.utils import translate_text
+from ner_v1.language_utilities.constant import TRANSLATED_TEXT
+
+
+class BaseDetector(object):
+    """
+    This class is the base class from which will be inherited by individual detectors. It primarily contains the
+    priority flow of structured value, message and fallback value and is also use to extend language support for
+    detectors using translation
+    
+    Attributes:
+        supported_languages (set): Set of languages supported by a particular detector
+        source_language_script (str): ISO-639 language code of query text for detection
+        target_language_script (str): ISO-639 language code in which detector would process the query
+        translation_enabled (bool): Decides to either enable or disable translation API
+    """
+    __metaclass__ = abc.ABCMeta
+
+    supported_languages = set()
+    source_language_script = ENGLISH_LANG
+    target_language_script = ENGLISH_LANG
+    translation_enabled = False
```

Comment:
I don't see the point of keeping these at class level. Why not make instance attributes, that way they will be overridable